src/components/pages/devices/flyouts/deviceNew/deviceNew.js

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React from 'react';
+
+import { IoTHubManagerService } from 'services';
+import {
+  copyToClipboard,
+  LinkedComponent,
+  svgs
+} from 'utilities';
+import { Svg } from 'components/shared/svg/svg';
+import {
+  Btn,
+  BtnToolbar,
+  ErrorMsg,
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  FormSection,
+  Indicator,
+  SectionDesc,
+  SectionHeader,
+  SummaryCount,
+  SummarySection
+} from 'components/shared';
+
+import './deviceNew.css';
+
+const DeviceDetail = ({ label, value }) => (
+  <FormSection className="device-detail">
+    <SectionHeader>{label}</SectionHeader>
+    <div className="device-detail-contents">
+      <div className="device-detail-value">{value}</div>
+      <Svg className="copy-icon" path={svgs.copy} onClick={() => copyToClipboard(value)} />
+    </div>
+  </FormSection>
+);
+
+const DeviceConnectionString = ({ label, deviceId, hostName, sharedAccessKey }) => (
+  <DeviceDetail label={label} value={`HostName=${hostName};DeviceId=${deviceId};SharedAccessKey=${sharedAccessKey}`} />
+);
+
+
+const ProvisionedDevice = ({ device, t }) => {
+  const { id, ioTHubHostName: hostName, authentication: { primaryKey }, authentication: { secondaryKey } } = device;
+
+  return (
+    <div>
+      <pre>{JSON.stringify(device, null, 2)}</pre>
+
+      <DeviceDetail label={t('devices.new.deviceId')} value={id} />
+      <DeviceDetail label={t('devices.new.primaryKey')} value={primaryKey} />
+      <DeviceDetail label={t('devices.new.secondaryKey')} value={secondaryKey} />
+      <DeviceConnectionString label={t('devices.new.primaryKeyConnection')} deviceId={id} hostName={hostName} sharedAccessKey={primaryKey} />
+      <DeviceConnectionString label={t('devices.new.secondaryKeyConnection')} deviceId={id} hostName={hostName} sharedAccessKey={secondaryKey} />
+    </div>
+  );
+};
+
+export class DeviceNew extends LinkedComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isPending: false,
+      error: undefined,
+      successCount: 0,
+      changesApplied: false,
+      formData: {
+        count: 1
+      },
+      provisionedDevice: {}
+    };
+
+    this.countLink = this.linkTo('count');
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  provisionDevices = () => {
+    this.setState({ isPending: true });
+
+    console.log(this.state);
+    console.log(this.countLink);
+
+    //TODO: Create the form to enable user input.
+    this.subscription = IoTHubManagerService.provisionDevice({ "Id": "", "Authentication": {}, "IsSimulated": false })
```

Comment:
Just like we have models that map from service response to UI model, we want to have models that from to request objects. You need to make on here. Also use single quotes.

Reply:
Yes, this is temporary... since this is just a super basic implementation for now.